packages/medusa/src/api/routes/admin/products/index.ts

```diff
@@ -132,38 +132,168 @@ export const defaultAdminProductFields: (keyof Product)[] = [
 
 export const defaultAdminGetProductsVariantsFields = ["id", "product_id"]
 
+/**
+ * @schema AdminProductsDeleteOptionRes
+ * type: object
+ * properties:
+ *   option_id:
+ *     type: string
+ *     description: The ID of the deleted Product Option
+ *   object:
+ *     type: string
+ *     description: The type of the object that was deleted.
+ *     default: option
+ *   deleted:
+ *     type: boolean
+ *     description: Whether or not the items were deleted.
+ *     default: true
+ *   product:
+ *     $ref: "#/components/schemas/Product"
+ */
 export type AdminProductsDeleteOptionRes = {
   option_id: string
   object: "option"
   deleted: boolean
   product: Product
 }
 
+/**
+ * @schema AdminProductsDeleteVariantRes
+ * type: object
+ * properties:
+ *   variant_id:
+ *     type: string
+ *     description: The ID of the deleted Product Variant.
+ *   object:
+ *     type: string
+ *     description: The type of the object that was deleted.
+ *     default: product-variant
+ *   deleted:
+ *     type: boolean
+ *     description: Whether or not the items were deleted.
+ *     default: true
+ *   product:
+ *     $ref: "#/components/schemas/Product"
+ */
 export type AdminProductsDeleteVariantRes = {
   variant_id: string
   object: "product-variant"
   deleted: boolean
   product: Product
 }
 
+/**
+ * @schema AdminProductsDeleteRes
+ * type: object
+ * properties:
+ *   id:
+ *     type: string
+ *     description: The ID of the deleted Product.
+ *   object:
+ *     type: string
+ *     description: The type of the object that was deleted.
+ *     default: product
+ *   deleted:
+ *     type: boolean
+ *     description: Whether or not the items were deleted.
+ *     default: true
+ */
 export type AdminProductsDeleteRes = {
   id: string
   object: "product"
   deleted: boolean
 }
 
+/**
+ * @schema AdminProductsListRes
+ * type: object
+ * properties:
+ *   products:
+ *     type: array
+ *     items:
+ *       $ref: "#/components/schemas/Product"
+ *   count:
+ *     type: integer
+ *     description: The total number of items available
+ *   offset:
+ *     type: integer
+ *     description: The number of items skipped before these items
+ *   limit:
+ *     type: integer
+ *     description: The number of items per page
+ */
 export type AdminProductsListRes = PaginatedResponse & {
   products: (PricedProduct | Product)[]
```

Comment:
Would it make sense to create a schema type for `PricedProduct` and include it here? We have an issue on linear about showing the pricing fields of variants since they're important during development. Creating a schema type for it would solve that issue. It could also be done as part of another PR.

Reply:
Yes, I believe it would make sense for PricedProduct to have its own @schema. I think I would not include it in this PR and I would address it once the Linear issue is prioritized to be worked on. Send me the issue number in order for me track it.